crates/ide_ssr/src/from_comment.rs

```diff
@@ -0,0 +1,26 @@
+//! This module allows building an SSR MatchFinder by parsing the SSR rule
+//! from a comment.
+
+use ide_db::{
+    base_db::{FilePosition, FileRange},
+    RootDatabase,
+};
+use syntax::ast::{self, AstToken};
+
+use crate::{MatchFinder, SsrError};
+
+pub fn ssr_from_comment<'db>(
+    db: &'db RootDatabase,
+    frange: FileRange,
+    comment: &ast::Comment,
+) -> Result<MatchFinder<'db>, SsrError> {
```

Comment:
Is this the interface we want? 

Should the caller pass in `ast::Comment` or just the `FileRange` and let `ssr_from_comment` try to find a comment? If we do that, the return type would need to change though, either to be more complex, or to just drop the errors and return `Option<MatchFinder>`. 

This also relates to how we test this. I think I can see a nice way to test the code in the `ide_ssr` crate. But as for the code in the `ide` crate, I haven't yet found the test infrastructure to test `add_ssr_assist`. Perhaps I should create this infrastructure based on the `ide_assists` crate?

Reply:
Yeah, I think it's better that the caller passes just the range. 

I don't think we need any kind of infrastructure here -- just a single test that it works would be enough.

Reply:
I've updated this so the caller passes the range. I forgot to mention there was one other reason I wasn't sure about that approach - it means `ssr_from_comment` also has to return the `TextRange` of the comment so it can be set as the target of the assist. But I suppose that is reasonable because it allows this to be expanded in the future to handle something like multi-line comments.